### PR TITLE
Refunds

### PR DESCRIPTION
**Pull request by AdrianB93**
Support transaction refunds on the billable and the invoice.

Tests included. Master and 2.0 appear out of sync, are there any contribution guides I should look at for this project? I'd have asked earlier but issues are not enabled.

### CODE REVIEW

File: src/Invoice.php [hunk 1]
```diff
@@ -144,6 +144,16 @@ public function hasDiscount()
...
     }
 
+    /**
+     * Determine if the invoice is for a refunded transaction.
+     *
+     * @return bool
+     */
+    public function isRefund()
+    {
+        return !! $this->transaction->refundedTransactionId;
```
Comment (taylorotwell) [inline src/Invoice.php]: This syntax is a bit confusing at first glance. Another approach?

Comment (AdrianB93) [reply src/Invoice.php]: Changed `!!` to `(bool)`

**Comment by AdrianB93** [comment]
@taylorotwell, I checked the stripe version of cashier, I'm unsure if the invoice blade shows the credited amount on the invoice, or indicate the transaction is a credit note.

For this PR, I changed the titles to "Credit Note" instead of "Invoice", and if `isRefund`, it will say "Credit for invoice number `REFUNDED_TRANSACTION_ID`" as the line item.

**Comment by taylorotwell** [comment]
Please re-submit this to the 2.0 branch.

**Pull request closed without merging**